ODIN_II/parse_odin_stats.py

```diff
@@ -0,0 +1,948 @@
+        print("\t-Will generate same ratio for certain core stats, otherwise must customize your own threshold file with metrics you want and ratio")
+        print("\t-Format is metric_name: <ratio>\nmetric_name2: <ratio>")
+        print("-------------------------------------")
+        print("SPECIFY THRESHOLD FILE:")
+        print("\tInclude [-t <threshold_filename>] to either compare mode before input/output to include threshold file to test against")
+        print("-------------------------------------")
+        print("SIMPLE_MODE:")
+        print("Take output of stat_gen as input, parse it and place results as one line in csv/tsv")
+        print("e.g.\t[./parse_odin_stats.py <input.log> <output.csv>]\n")
+        
+    elif(code == 1):
+        print("Nothing to compare, use -mc mode")
+
+    elif(code == 2):
+        print("\tNo inputs and/or outputs specified, expecting [./exec <-m/mc> (optional)<--flags/-f> (opt)<--concise> <output> <input-1> ... <input-n>]")
+
+    elif(code == 3):
+        print("Incorrect threshold generator usage. Acceptable patterns below:")
+        print("\n1) [./exec -gen] -> yields thresholds.txt with every value as 0.15 ('-g' also accepted)")
+        print("2) [./exec -gen <ratio>] (e.g. -gen 0.25) -> specify ratio to generate thresholds.txt with custom ratio")
+        print("3) [./exec -gen <ratio> <outputfilename.txt>] -> yields specified threshold file name with specified ratio (custom name requires custom ratio)\n")
+
+    elif(code==4):
+        print("No input specified")
+    elif(code==5):
+        print("Formatting error: check log file")
+    elif(code==6):
+        print("Wrong number of arguments, expecting [./exec -c <input.log> <input2.log> (opt)<output.csv>]")
+    elif(code==7):
+        # this is our only mechanism against the possibility of overwriting the log file, important that input files are .log
+        print("\nImproper output format, output file must not begin with '-' or end in .log, use no args or -h/--help for proper format")
+
+
+def size_to_MiB(size_in):
+    size_in = format_token(size_in)
+
+    base_size = 1.0
+    if 'b' in size_in:
+        base_size = 8.0
+
+    size_in = size_in.replace('i','')
+    size_in = size_in.replace('I','')
+    size_in = size_in.replace('b','')
+    size_in = size_in.replace('B','')
+
+    if 'G' in size_in or 'g' in size_in:
+        size_in = str(assert_float(size_in.replace('G','').replace('g','')) / 1024.0 / 1024.0 / base_size)
+
+    elif 'M' in size_in or 'm' in size_in:
+        size_in = str(assert_float(size_in.replace('M','').replace('m','')) / 1024.0 / base_size)
+
+    elif 'K' in size_in or 'k' in size_in:
+        size_in = str(assert_float(size_in.replace('K','').replace('k','')) / 1024.0 / base_size)
+
+    #we are now in byte
+    return str(assert_float(size_in) * 1024.0 * 1024.0)
+
+def contains(line_in, on_tokens):
```

Comment:
I don't know that this will ever come up as an issue but this matches the on_tokens in any order on the line, If this is not intended I would revise it such that they stay in order, match first token, match second token on substring after first, etc...